test/unit/files/test_basespace.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+import pytest
+
+from galaxy.files import ConfiguredFileSources, ConfiguredFileSourcesConfig
+from ._util import (
+    assert_realizes_as,
+    find,
+    user_context_fixture,
+)
+SCRIPT_DIRECTORY = os.path.abspath(os.path.dirname(__file__))
+FILE_SOURCES_CONF = os.path.join(SCRIPT_DIRECTORY, "basespace_file_sources_conf.yml")
+
+skip_if_no_basespace_access_token = pytest.mark.skipif(
+    not os.environ.get('GALAXY_TEST_BASESPACE_CLIENT_ID')
+    or not os.environ.get('GALAXY_TEST_BASESPACE_CLIENT_SECRET')
+    or not os.environ.get('GALAXY_TEST_BASESPACE_ACCESS_TOKEN')
+    or not os.environ.get('GALAXY_TEST_BASESPACE_TEST_FILE_PATH'),
+    reason="GALAXY_TEST_BASESPACE_CLIENT_ID and related vars not set"
+)
+
+
+@skip_if_no_basespace_access_token
```

Comment:
The added unit tests are all using the same test structure, can you parameterize them ?

Reply:
That's something that's generally the case for many file source tests. Maybe a refactoring run that should be done outside of this PR?

Reply:
Can you at least push `_configured_file_sources ` into the helper ? That is literally the same in all those tests.

Reply:
Have refactored the tests and helpers. Can you take a look?